### PR TITLE
Add snap mini corr

### PR DESCRIPTION
**Pull request by mkolopanis**
Adds a mini data catcher script to save snap spectra to a UVH5 file.
## Description

Reads the `status:snap:*` keys in redis and compiles them into a UVData object to save to disk.
## Motivation and Context

Danny asked for this.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Schema change (any change to the SQL tables)
- [ ] New feature without schema change (non-breaking change which adds functionality)
- [ ] Change associated with a change in redis structure
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Version change
- [ ] Build or continuous integration change
- [x] Other

## Checklist:

Other:
- [x] My code follows the code style of this project.
- [x] I understand the updates required onsite (detailed in the readme) and I will make those
changes when this is merged.
- [x] Unit tests pass **on site** (This is a critical check, CI can differ from site).
- [ ] I have updated the [CHANGELOG](https://github.com/HERA-Team/hera_mc/blob/main/CHANGELOG.md).

### CODE REVIEW

**Comment by mkolopanis** [comment]
@bhazelton I think this is worth a look. I'm open to any ideas.

we still have a problem were sometimes while loop thinks it is ready to trigger but only after 200ms. I think this is due to how snap spectra only update on a ~10 min timescale and even then only roll out in stages.

File: scripts/hera_mini_snap_data_catcher.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+                # Nblts
+                time_array = []
+                ant_array = []
+
+                downtime = 0
+                file_len = 0
+
+                while downtime < args.max_downtime and file_len < args.max_file_len:
+                    time.sleep(0.01)
+                    # do the thing
```
Comment (bhazelton) [inline scripts/hera_mini_snap_data_catcher.py:133]: remove this line or make it more helpful

File: scripts/hera_mini_snap_data_catcher.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+                                # Time has not progressed yet.
+                                # Just sit tight
+                                continue
+                            last_time_mapping[snap_hostname] = timestamp
+
+                        auto_corr = status["autocorrelation"]
+
+                        if data_array is None:
+                            data_array = np.zeros(
+                                (0, auto_corr.shape[0], 2), dtype=np.float32
```
Comment (bhazelton) [inline scripts/hera_mini_snap_data_catcher.py:174]: Why is the first axis length 0?

**Review by bhazelton — COMMENTED** [review]
Looking good!

File: scripts/hera_mini_snap_data_catcher.py (hunk 2 above)
Comment (mkolopanis) [reply scripts/hera_mini_snap_data_catcher.py:174]: The 0 in the shape acts as a placeholder axis so we can accumulate data into it until we reach either timeout condition or a hour of data. I didn't want to necessarily assume how long a data file would get so I let it grow.

**Review by bhazelton — APPROVED** [review]
I think this is ready. Eventually we might want to put more of the logic in functions that can be tested, but that's hard right now and we should just get this in.

**Pull request merged**